src/app/system_test.go

```diff
@@ -406,3 +410,93 @@ func TestSystem_TestProcListToRun(t *testing.T) {
 		}
 	})
 }
+
+func TestSystem_TestProcListShutsDownInOrder(t *testing.T) {
+	fixture1 := filepath.Join("..", "..", "fixtures-code", "process-compose-shutdown-inorder.yaml")
+	t.Run("Single Proc with deps", func(t *testing.T) {
+
+		project, err := loader.Load(&loader.LoaderOptions{
+			FileNames: []string{fixture1},
+		})
+		if err != nil {
+			t.Errorf(err.Error())
+			return
+		}
+		numProc := len(project.Processes)
+		runner, err := NewProjectRunner(&ProjectOpts{
+			project:           project,
+			processesToRun:    []string{},
+			mainProcessArgs:   []string{},
+			isOrderedShutDown: true,
+		})
+		if err != nil {
+			t.Errorf(err.Error())
+			return
+		}
+		if len(runner.project.Processes) != numProc {
+			t.Errorf("should have %d processes", numProc)
+		}
+		for name, proc := range runner.project.Processes {
+			if proc.Disabled {
+				t.Errorf("process %s is disabled", name)
+			}
+		}
+		file, err := os.CreateTemp("/tmp", "pc_log.*.log")
+		defer os.Remove(file.Name())
+		project.LogLocation = file.Name()
+		project.LoggerConfig = &types.LoggerConfig{
+			FieldsOrder:     []string{"message"},
+			DisableJSON:     true,
+			TimestampFormat: "",
+			NoMetadata:      true,
+			FlushEachLine:   true,
+		}
+		go runner.Run()
+		time.Sleep(10 * time.Millisecond)
+		states, err := runner.GetProcessesState()
+		if err != nil {
+			t.Errorf(err.Error())
+			return
+		}
+		want := 3
+		if len(states.States) != want {
+			t.Errorf("len(states.States) = %d, want %d", len(states.States), want)
+		}
+
+		time.Sleep(10 * time.Millisecond)
+		err = runner.ShutDownProject()
+		if err != nil {
+			t.Errorf(err.Error())
+			return
+		}
+		runningProcesses := 0
+		for _, proc := range runner.runningProcesses {
```

Comment:
Race condition here.
You can replace this block with `runner.GetProcessesState()` (plural)

Reply:
Fixed here:
https://github.com/F1bonacc1/process-compose/commit/27c3ab97973b78499a38d50fbd41d1d4514224eb